src/ucp/wireup/select.c

```diff
@@ -1063,6 +1056,23 @@ ucs_status_t ucp_wireup_select_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
                                                           lane_descs,
                                                           key->num_lanes);
 
+    /* add to map first UCP_MAX_OP_MDS fastest MD's */
+    for (i = 0, rma_bw_md_count = 0;
+         key->rma_bw_lanes[i] != UCP_NULL_LANE && rma_bw_md_count < UCP_MAX_OP_MDS; i++) {
+        lane = key->rma_bw_lanes[i];
+        rsc_index = lane_descs[lane].rsc_index;
+        md_index  = worker->context->tl_rscs[rsc_index].md_index;
+
+        /* Pack remote key only if needed for RMA.
+         * FIXME a temporary workaround to prevent the ugni uct from using rndv. */
+        if ((context->tl_mds[md_index].attr.cap.flags & UCT_MD_FLAG_NEED_RKEY) &&
+            !(strstr(context->tl_rscs[rsc_index].tl_rsc.tl_name, "ugni"))) {
+            key->rma_bw_md_map |= UCS_BIT(md_index);
```

Comment:
missed that code earlier.
shouldn't we check if bit is already set, to count correctly?

Reply:
maybe use ucs_count_one_bits(key->rma_bw_md_map) instead of rma_bw_md_count

Reply:
fixed